From 8cc0b589d8f9d0b4f4c7116e5b566441e169ee51 Mon Sep 17 00:00:00 2001
From: Shiyong Li <a22381@motorola.com>
Date: Tue, 20 Oct 2015 15:58:15 -0700
Subject: [PATCH 391/959] IKSWL-39370 arm/dts: 8952: Add dummy panel for bare
 board

- Add dummy video panel configs.
- Disable "qcom,mdss-fb-splash-logo-enabled" property to make dummy
video panel work.

Change-Id: I74f73a687d0f46c3cb27fc6c0bee8165d254d348
Signed-off-by: Shiyong Li <a22381@motorola.com>
Reviewed-on: http://gerrit.mot.com/793083
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Bang Nguyen <bangnguyen@motorola.com>
Reviewed-by: Zachary Crosby <zac@motorola.com>
Reviewed-by: Vadim A Blaker <vblaker@motorola.com>
Reviewed-by: Joseph Swantek <jswantek@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 .../dts/qcom/dsi-panel-mot-dummy-qhd-video.dtsi    | 56 ++++++++++++++++++++++
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi    |  5 ++
 2 files changed, 61 insertions(+)
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-dummy-qhd-video.dtsi

diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-dummy-qhd-video.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-dummy-qhd-video.dtsi
new file mode 100644
index 00000000000..afa4cf46d31
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-dummy-qhd-video.dtsi
@@ -0,0 +1,56 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&mdss_mdp {
+	mot_dummy_vid_qhd: qcom,mdss_dsi_mot_dummy_qhd_video {
+		qcom,mdss-dsi-panel-name = "mipi_mot_video_dummy_qhd";
+		qcom,mdss-dsi-panel-type = "dsi_video_mode";
+		qcom,mdss-dsi-panel-framerate = <60>;
+		qcom,mdss-dsi-virtual-channel-id = <0>;
+		qcom,mdss-dsi-stream = <0>;
+		qcom,mdss-dsi-panel-width = <540>;
+		qcom,mdss-dsi-panel-height = <960>;
+		qcom,mdss-pan-physical-width-dimension = <54>;
+		qcom,mdss-pan-physical-height-dimension = <95>;
+		qcom,mdss-dsi-h-front-porch = <32>;
+		qcom,mdss-dsi-h-back-porch = <4>;
+		qcom,mdss-dsi-h-pulse-width = <32>;
+		qcom,mdss-dsi-h-sync-skew = <0>;
+		qcom,mdss-dsi-v-back-porch = <16>;
+		qcom,mdss-dsi-v-front-porch = <15>;
+		qcom,mdss-dsi-v-pulse-width = <1>;
+		qcom,mdss-dsi-bpp = <24>;
+		qcom,mdss-dsi-underflow-color = <0xff>;
+		qcom,mdss-dsi-border-color = <0>;
+
+		qcom,mdss-dsi-on-command-state = "dsi_lp_mode";
+		qcom,mdss-dsi-off-command-state = "dsi_lp_mode";
+		qcom,mdss-dsi-h-sync-pulse = <0>;
+		qcom,mdss-dsi-traffic-mode = "non_burst_sync_event";
+		qcom,mdss-dsi-bllp-eof-power-mode;
+		qcom,mdss-dsi-bllp-power-mode;
+		qcom,mdss-dsi-lane-0-state;
+		qcom,mdss-dsi-lane-1-state;
+		qcom,mdss-dsi-tx-eot-append;
+		qcom,mdss-dsi-panel-timings =
+				[82 1c 12 00 40 42 16 1e 14 03 04 00];
+		qcom,mdss-dsi-t-clk-post = <0x1b>;
+		qcom,mdss-dsi-t-clk-pre = <0x1b>;
+		qcom,mdss-dsi-dma-trigger = "trigger_sw";
+		qcom,mdss-dsi-mdp-trigger = "none";
+		qcom,mdss-dsi-reset-sequence = <1 0>;
+
+		qcom,mdss-dsi-bl-min-level = <1>;
+		qcom,mdss-dsi-bl-max-level = <255>;
+		qcom,mdss-dsi-bl-pmic-control-type = "bl_ctrl_dcs";
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index c1fe3a67de3..909eaa01f58 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -14,6 +14,7 @@
 #include "msm8952.dtsi"
 #include "msm8952-pinctrl.dtsi"
 #include "msm8952-camera-sensor-moto.dtsi"
+#include "dsi-panel-mot-dummy-qhd-video.dtsi"
 #include <moto-mem-reserve.h>
 
 / {
@@ -684,6 +685,10 @@
 	/delete-property/ ibb-supply;
 };
 
+&mdss_fb0 {
+	/delete-property/ qcom,mdss-fb-splash-logo-enabled;
+};
+
 &i2c_3 { /* BLSP1 QUP3 */
 	synaptics@20 {
 		compatible = "synaptics,synaptics_dsx_i2c";
-- 
2.11.0

